### PR TITLE
close #9, simple map

### LINKED ISSUE #9: simple Node-Edge Map for DFS Testing
Write a depth-first search algorithm and test it on a very simple map with **nodes and edges that you create yourselves.**

Submit:
The A* algorithm IntelliJ project along with your simple map. You do not need to submit the depth-first search pathfinding program.

### PR DESCRIPTION
**Pull request by cvkittler**
finished weighted graph class

### CODE REVIEW

File: src/main/java/edu/wpi/teamname/edge.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    this.edgeID = _edgeID;
+    this.startNode = _startNode;
+    this.endNode = _endNode;
+    this.weight = calcWeight(_startNode, _endNode);
+  }
+
+  private double calcWeight(node _startNode, node _endNode) {
+    // calculate the rise and run
+    double dx = Math.abs(_startNode.xcoord - _endNode.xcoord);
+    double dy = Math.abs(_startNode.ycoord = _endNode.ycoord);
```
Comment (Kohmei358) [inline src/main/java/edu/wpi/teamname/edge.java:19]: That should be a "-" not "="

**Review by Kohmei358 — COMMENTED** [review]
I think we should hold on merging into master until all the algo stuff is done but looks great so far!

File: src/main/java/edu/wpi/teamname/weightedGraph.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    this.numNodes = nodeList.length;
+    this.numEdges = edgeList.length;
+
+    adjacencyMatrix = new double[numNodes][numNodes];
+
+    for (int i = 0; i < numEdges; i++) {
+      this.adjacencyMatrix[edgeList[i].startNode.nodeID][edgeList[i].endNode.nodeID] =
+          edgeList[i].weight;
+      this.adjacencyMatrix[edgeList[i].endNode.nodeID][edgeList[i].startNode.nodeID] =
+          edgeList[i].weight;
```
[the remark this answers is not in the document]
Comment (Kohmei358) [reply src/main/java/edu/wpi/teamname/weightedGraph.java:18]: Actually, I was reading floor map entry instructions and found these instructions:

> nodeID – follow the naming convention below for new nodes
> 1.    All characters are in uppercase
> 2.    First character is your team letter
> 3.    Next 4 characters is the node type.
> 4.    Next 3 characters is the number of that node type for the floor it is on. Use leading zeros if your number is less than one hundred. If it is an elevator use two zeros followed by the elevator letter. There are three elevators in the Faulkner Hospital – one towards the top of the map and two close to one another towards the bottom of the map. The top one is elevator X, the bottom left is elevator Y, and the bottom right is elevator Z.
> 5.    The last two characters is the floor. Use one of the following: 01, 02, 03, 04, 05, 06, or 07
> 
> The edgeID is simply the two connected nodeID’s with an underscore separating the two IDs. Note that only one direction is provided in the raw data. You should be able to either programmatically generate the two edges for both directions or have your algorithm assume bi-directionality. In other words, if you have nodes A and B, the raw data provides edgeID A_B, starting node A, ending node B. You can write Java code to create the reverse direction: edgeID B_A, starting node B, ending node A. 

Are you planning to change the map structure later when we have real data? Or do you think it should be switched now?

File: src/main/java/edu/wpi/teamname/node.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package edu.wpi.teamname;
+
+public class node {
```
Comment (aimiebell) [inline src/main/java/edu/wpi/teamname/node.java:3]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).

**Pull request closed without merging**